Session/Utilities/BackgroundPoller.swift

```diff
@@ -50,13 +50,58 @@ public final class BackgroundPoller : NSObject {
                         let job = MessageReceiveJob(data: data, serverHash: json["hash"] as? String, isBackgroundPoll: true)
                         return job.execute()
                     }
-                    
                     // Now that the MessageReceiveJob's have been created we can update the `lastMessageHash` value
-                    SnodeAPI.updateLastMessageHashValueIfPossible(for: snode, associatedWith: publicKey, from: lastRawMessage)
+                    SnodeAPI.updateLastMessageHashValueIfPossible(for: snode, namespace: SnodeAPI.defaultNamespace, associatedWith: publicKey, from: lastRawMessage)
                     
                     return when(fulfilled: promises) // The promise returned by MessageReceiveJob never rejects
                 }
             }
         }
     }
+    
+    private static func getClosedGroupMessages(for publicKey: String) -> Promise<Void> {
```

Comment:
The handling of the promise response in here seems to be the same between both promises - it would be good if we could avoid duplicating the code by just setting the promise based on the condition and handling the diff the same:
```
private static func getClosedGroupMessages(for publicKey: String) -> Promise<Void> {
    return SnodeAPI.getSwarm(for: publicKey).then(on: DispatchQueue.main) { swarm -> Promise<Void> in
        guard let snode = swarm.randomElement() else { throw SnodeAPI.Error.generic }

        return attempt(maxRetryCount: 4, recoveringOn: DispatchQueue.main) {
            var promises: [Promise<Void>] = []
            let promise: Promise<JSON>

            if SnodeAPI.hardfork <= 19, SnodeAPI.softfork == 0 {
                promise = SnodeAPI.getRawClosedGroupMessagesFromDefaultNamespace(from: snode, associatedWith: publicKey)
            }
            else {
                promise = SnodeAPI.getRawMessages(from: snode, associatedWith: publicKey, authenticated: false)
            }

            promise.then(on: DispatchQueue.main) { rawResponse -> Promise<Void> in
                let (messages, lastRawMessage) = SnodeAPI.parseRawMessagesResponse(rawResponse, from: snode, associatedWith: publicKey)
                let promises = messages.compactMap { json -> Promise<Void>? in
                    // Use a best attempt approach here; we don't want to fail the entire process if one of the
                    // messages failed to parse.
                    guard let envelope = SNProtoEnvelope.from(json),
                        let data = try? envelope.serializedData() else { return nil }
                    let job = MessageReceiveJob(data: data, serverHash: json["hash"] as? String, isBackgroundPoll: true)
                    return job.execute()
                }
                // Now that the MessageReceiveJob's have been created we can update the `lastMessageHash` value
                SnodeAPI.updateLastMessageHashValueIfPossible(for: snode, namespace: SnodeAPI.defaultNamespace, associatedWith: publicKey, from: lastRawMessage)

                return when(fulfilled: promises) // The promise returned by MessageReceiveJob never rejects
            }
            promises.append(promise)
            return when(resolved: promises).map { _ in }
        }
    }
}
```

Reply:
That's true. I've cleaned up the code. :)